src/ncp/ncp_spinel.hpp

```diff
@@ -49,6 +49,28 @@
 namespace otbr {
 namespace Ncp {
 
+/**
+ * This interface is an observer to subscribe the network properties from NCP.
+ *
+ */
+class PropsObserver
```

Comment:
Is it necessary to define an interface class for this?

Reply:
This has a key advantage: The user of `NetworkProperties` can only see and use the Get methods and cannot set the values. Otherwise we need to introduce 'friend', which is not a preferred way.